From f8ed134e6fcd160675cf46b7d7b6868c7aa59ee0 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Thu, 13 Feb 2020 09:58:54 +0200
Subject: [PATCH] [SW-8229] goya2: check security bypass bootstrap

If security verification efuse isn't set we can still
toggle bypass of security via bootstrap. Only check it
while efuse isn't set, once efuse is blown bootstrap is
ignored

Change-Id: I3dd1e5e4f35f44087419619b64f83c65f7a18e0e
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/secureboot.c b/common/secureboot.c
index c1fec8a..f84a851 100644
--- a/common/secureboot.c
+++ b/common/secureboot.c
@@ -49,12 +49,15 @@
 	uint32_t fw_version = base_readl((uint32_t)test_phase0b_data);
 
 	/* TODO: remove any global or ROM specific rutines */
-	/* TODO: add depenancie on validate starp if efuse isn't set */
-	/* TODO: gpio46 is validate strap -  0 is to validate*/
+	/* check efuse for security validation enable */
 	if ((arcboot_rom_global.config_fuse &
-	     EFUSE_SYSCFG_ENABLE_VALIDATION_MASK) == 0) {
-		return SECUREBOOT_VERIFY_SUCCESS;
-	}
+	     EFUSE_SYSCFG_ENABLE_VALIDATION_MASK) == 0)
+		/* if efuse isn't set, check status of security bypass
+		 * boot strap if enabled bypass security validation.
+		 */
+		if ((regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H) &
+		     PSOC_GLOBAL_CONF_BOOT_STRAP_PINS_H_SECURITY_BYPASS_MASK))
+			return SECUREBOOT_VERIFY_SUCCESS;
 
 	/* check minimal valid security FW version */
 	if (security_check_valid_fw_version(fw_version) !=
